### PR TITLE
common: add support for fourth and fifth pointing-device buttons

### LINKED ISSUE #6043: Add support for five pointing-device buttons
Currently, `PointerEventButton` only supports three specific buttons. Others like the fourth and fifth are grouped together under `other`, rendering you unable to distinguish them. The Windows OS has built-in support for up to five buttons. It would be great to have access to all of them in Slint to support functionality like "back" and "forward". I propose the new variants `fourth` and `fifth`.

### PR DESCRIPTION
**Pull request by crai0**
Closes #6043

### CODE REVIEW

**Comment by crai0** [comment]
I was unsure about whether I should touch the `testing` backend or if I should add tests. Please let me know if anything else needs to be done here.

**Comment by Enyium** [comment]
Why not, I'd say.

For the record: Windows [calls](https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-xbuttondown) them XBUTTON1 and XBUTTON2.

But GPT-4o agrees that "back" and "forward" is the most suitable.

Internet browsers, file browsers and text editors like VS Code all use these buttons for back and forward functionality.

**Comment by crai0** [comment]
I adjusted the naming accordingly and also found an example (`examples/cpp/platform_native`) where the buttons could be added.

**Comment by Enyium** [comment]
@crai0: As you're still editing this, I think `winit::event::MouseButton::Other(_) => PointerEventButton::Other,` should be at the end, and `Other,` in `enum PointerEventButton {` also.

**Comment by crai0** [comment]
I thought that I could be clever and do these last few changes on mobile but that was foolish. I will finish this up tomorrow.

File: internal/common/enums.rs [hunk 1]
```diff
@@ -147,15 +147,19 @@ macro_rules! for_each_enums {
             /// typically on a mouse or a pencil.
             #[non_exhaustive]
             enum PointerEventButton {
-                /// A button that is none of left, right or middle. For example
-                /// this is used for a fourth button on a mouse with many buttons.
-                Other,
```
Comment (ogoffart) [inline internal/common/enums.rs:152]: Other need to stay the first because the first value is implicitly the default

**Comment by crai0** [comment]
Okay, this should be done now.

**Review by ogoffart — APPROVED** [review]
Thanks a lot for the contribution

**Pull request merged**